File: clients/javascript/src/models/kvGetIn.ts
// this file is @generated
import {
    type Consistency,
    ConsistencySerializer,
} from './consistency';

export interface KvGetIn {
    namespace?: string | null;
    consistency?: Consistency;
}

export interface KvGetIn_ {
    namespace?: string | null;
    key: string;
    consistency?: Consistency;
}

export const KvGetInSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): KvGetIn_ {
        return {
            namespace: object['namespace'],
            key: object['key'],
            consistency: object['consistency'] != null ? ConsistencySerializer._fromJsonObject(object['consistency']): undefined,
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: KvGetIn_): any {
        return {
            'namespace': self.namespace,
            'key': self.key,
            'consistency': self.consistency != null ? ConsistencySerializer._toJsonObject(self.consistency) : undefined,
        };
    }
}

File: z-clients/python/coyote/models/kv_get_in.py
# this file is @generated
import typing as t

from pydantic import BaseModel

from .consistency import Consistency


class KvGetIn(BaseModel):
    namespace: t.Optional[str] = None

    consistency: t.Optional[Consistency] = None


class _KvGetIn(BaseModel):
    namespace: t.Optional[str] = None

    key: str

    consistency: t.Optional[Consistency] = None
